dimension/dimension.go

```diff
@@ -95,67 +111,106 @@ func (s *Store) GetUnique(w http.ResponseWriter, r *http.Request) {
 	log.Debug("get dimension values", log.Data{"instance": id})
 }
 
-// Add represents adding a dimension to a specific instance
-func (s *Store) Add(w http.ResponseWriter, r *http.Request) {
+// AddHandler represents adding a dimension to a specific instance
+func (s *Store) AddHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
 	vars := mux.Vars(r)
-	id := vars["id"]
+	instanceID := vars["id"]
+	logData := log.Data{"instance_id": instanceID}
+	auditParams := common.Params{"instance_id": instanceID}
 
-	// Get instance
-	instance, err := s.GetInstance(id)
+	statusCode, err := s.add(ctx, w, r, instanceID, logData)
 	if err != nil {
-		log.ErrorC("Failed to GET instance", err, log.Data{"instance": id})
-		handleErrorType(err, w)
+		s.Auditor.Record(ctx, PostDimensionsAction, audit.Unsuccessful, auditParams)
+
+		handleDimensionErr(ctx, err, statusCode, w, logData)
 		return
 	}
 
+	s.Auditor.Record(ctx, PostDimensionsAction, audit.Successful, auditParams)
+
+	log.InfoCtx(ctx, "added dimension to instance resource", logData)
+}
+
+func (s *Store) add(ctx context.Context, w http.ResponseWriter, r *http.Request, instanceID string, logData log.Data) (int, error) {
```

Comment:
Consider making this function strictly business logic only and not http related. 
* Consider replacing the `r` parameter with `option`. This would involve unmarshalling the option in the handler
* Consider removing the `int` return parameter, and using specific error's instead

(In my mind the business logic is what should be in a separate package from `api`, and that all the http specifics would be left in `api`)